galette/lib/Galette/Entity/ScheduledPayment.php

```diff
@@ -193,12 +193,12 @@ public function check(array $data): bool
         }
 
         $this
-            ->setContribution($data[Contribution::PK])
-            ->setPaymentType($data['id_paymenttype'])
+            ->setContribution((int)$data[Contribution::PK])
+            ->setPaymentType((int)$data['id_paymenttype'])
             ->setCreationDate($data['creation_date'] ?? date('Y-m-d'))
             ->setScheduledDate($data['scheduled_date'])
-            ->setAmount($data['amount'] ?? $this->contribution->amount)
-            ->setPaid($data['is_paid'] ?? false)
+            ->setAmount((float)$data['amount'] ?? $this->contribution->amount)
```

Comment:
I guess typehinting a non existing array entry will fail:
```suggestion
            ->setAmount(isset($data['amount']) ? (float)$data['amount'] : $this->contribution->amount)
```